MSAL/src/native_auth/controllers/reset_password/MSALNativeAuthResetPasswordController.swift

```diff
@@ -156,16 +156,16 @@ final class MSALNativeAuthResetPasswordController: MSALNativeAuthBaseController,
                 correlationId: context.correlationId()
             )
             stopTelemetryEvent(event, context: context, error: error)
-            MSALLogger.log(level: .error,
-                           context: context,
-                           format: "Redirect error in resetpassword/start request \(error.errorDescription ?? "No error description")")
+            MSALLogger.logPII(level: .error,
+                              context: context,
+                              format: "Redirect error in resetpassword/start request \(MSALLogMask.maskPII(error.errorDescription))")
```

Comment:
Can you please check with MSAL Apple team (or Sergei) if we still need to mask PII if we log using PII? I'd expect that if we notify that the log contains PII there is no need to mask the content. At least this is how native auth Android operates and it makes sense to me. I know I wrote differently in the PBI description, but this is something came later :/

Reply:
This sample code is to show what we talked today in the morning:

![Screenshot 2024-07-23 at 17 33 41](https://github.com/user-attachments/assets/df06f27a-20a5-4af8-8528-d081338ab5a8)

If the developer has the **default configuration** for MSAL:

`MSALGlobalConfig.loggerConfig.logMaskingLevel = .settingsMaskAllPII`

These are the resulting logs:

![Screenshot 2024-07-23 at 17 45 17](https://github.com/user-attachments/assets/bc0b374c-7a70-4114-9436-96f0e55ba55e)

In the first log the username is leaked. Note how in both logs the `containsPII` bool is false. This is because the `logMaskingLevel` is set to `.settingsMaskAllPII`.

===

Now let's configure the SDK with the most permissive masking level:

`MSALGlobalConfig.loggerConfig.logMaskingLevel = .settingsMaskSecretsOnly`

These are the resulting logs:

![Screenshot 2024-07-23 at 17 40 06](https://github.com/user-attachments/assets/422f7be8-d2ac-4018-9698-5c79605b0d0d)

Our masked log now shows the full errorDescription and the `containsPII` bool is true.

Reply:
Thanks for the clarification Diego